messaging/src/main/java/org/axonframework/common/ProcessUtils.java

```diff
@@ -49,4 +51,30 @@ public static void executeWithRetry(Runnable runnable, Predicate<RuntimeExceptio
 
         throw lastException;
     }
+
+    /**
+     * Executes an action, with potential retry in case the result is false. Exception handling should be taken care of
+     * within the action if needed.
+     *
+     * @param runnable      action to execute
+     * @param retryInterval time to wait between retries of the action
+     * @param maxTries      maximum number of times the action is invoked,
+     */
+    public static void executeUntilTrue(BooleanSupplier runnable, long retryInterval, long maxTries) {
+        AtomicLong totalTriesCounter = new AtomicLong();
+        boolean result = runnable.getAsBoolean();
+        while (!result) {
+            if (totalTriesCounter.incrementAndGet() >= maxTries){
+                throw new AxonRetriesExhaustedException(String.format(
```

Comment:
I'd would like to get rid of the `Axon` bit in the exception here. 
Granted, `RetriesExhaustedException` also isn't overly clear to users perhaps.

Reply:
Done with later suggestion.